### DISCUSSION TITLE
Can app detect which repo it was deployed from (npm or github)?

### DISCUSSION THREAD

**Discussion by jpage4500** [shannah/jdeploy#147] [Q&A]
I saw this post a while back:

>> You could use the jdeploy.app.version system property. Store somewhere, e.g. in Preferences, on launch. If it is different than the value you stored last time, then you know it has been updated.

It got me thinking if the app could tell which repo jdeploy was using. The idea is that I already check for updates in my app while it's running.. I could periodically check for new versions via one of the repos and let the user know they can update on a restart

(my app tends to be run once on a laptop and never restarted -- otherwise this wouldn't be necessary)

**Comment by shannah** [comment]
Nice idea.  Wouldn't be hard to implement.  I'd probably implement it as a separate pure java lib the you could include in your app if you wanted this functionality.

**Reply by charot-git** [reply]
How the lib for this? It's a good tool for deployed apps through jdeploy.
